poc/tenant-controller/pkg/controllers/tenants/controller.go

```diff
@@ -77,7 +98,16 @@ func (c *Controller) updateTenant(old, obj *tenantsapi.Tenant) {
 func (c *Controller) deleteTenant(obj *tenantsapi.Tenant) {
 	glog.Info("deleteTenant: %#v", obj)
 	deleteRBACForTenantCRD(obj)
-	// TODO delete namespace
+	// TODO add a full set of sanity checks in future before deleting
+
+	for n, _ := range c.tenantsList[obj.ObjectMeta.Name].Spec.Namespaces {
```

Comment:
It's not necessary to get namespaces from c.tenantsList. The current `obj` provides all namespaces already. That's the reason I mentioned currently there's no need for tenantsList.

Reply:
As mentioned above, there is a benefit to doing it this way but will change it if we decide to remove the tenantsList